src/main/scala-2.12/com/linagora/gatling/imap/protocol/command/SubscribeHandler.scala

```diff
@@ -0,0 +1,45 @@
+package com.linagora.gatling.imap.protocol.command
+
+import java.util.function.Consumer
+
+import akka.actor.{ActorRef, Props}
+import com.linagora.gatling.imap.protocol._
+import com.yahoo.imapnio.async.client.ImapAsyncSession
+import com.yahoo.imapnio.async.request.SubscribeFolderCommand
+import com.yahoo.imapnio.async.response.ImapAsyncResponse
+import io.gatling.core.akka.BaseActor
+
+import scala.collection.immutable.Seq
+
+object SubscribeHandler {
+  def props(session: ImapAsyncSession) = Props(new SubscribeHandler(session))
+}
+
+class SubscribeHandler(session: ImapAsyncSession) extends BaseActor {
+  override def receive: Receive = {
+    case Command.Subscribe(userId, mailbox) =>
+      logger.trace(s"SubscribeHandler for user : ${userId.value}, on actor ${self.path} responding to ${sender.path}")
+      context.become(waitCallback(sender()))
+
+      val responseCallback: Consumer[ImapAsyncResponse] = responses => {
+        import collection.JavaConverters._
+
+        val responsesList = ImapResponses(responses.getResponseLines.asScala.to[Seq])
+        logger.trace(s"On response for $userId :\n ${responsesList.mkString("\n")}")
+        self !  Response.SubscribeResponse(responsesList)}
+      val errorCallback: Consumer[Exception] = e => {
+        logger.error("SubscribeHandler command failed", e)
+      }
```

Comment:
Please propagate errors to the sender cf https://github.com/linagora/gatling-imap/pull/62/commits/e3765a35c48890ecd8b9f676a646e6b08da72b63

Here and in all your other handlers...